Interface.py

```diff
@@ -8,56 +8,60 @@
     #research "requests"", use "posts" and json, and pickle
         #pickle converts arguments into string, and trns into json
         #1: requests 2: pickle, 3: understand json
+
+    #verify security headers in the server, thats more of a complex task
 import requests
 import json
+import urllib.parse
 
 class Executor():
-    def __init__(self, host, port, path = None, params = None):
-        self.server_info = (host, port, path, params)
-        assert ((self.server_info[0] is not None) and (self.server_info[1] is not None))
-        self.url_str = "http://" + self.server_info[0] + ":" + self.server_info[1] + "/"
-        if self.server_info[2] is not None:
-            self.url_str += self.server_info[2]
-        if self.server_info[3] is not None:
-            self.url_str += "?" + self.server_info[3]
+    #delete this __init__ when I've finished server side implementation
+    def __init__(self, url, path = None, params = None):
+        self.url_str = url
+           
+        #recode this section into taking host and port as params
 
-        #server stores info of jobs in centralized location, so should manage encoding
+        print(self.url_str)
 
     def submit(self, fn, job_name, memory, *args, priority = 1, retries = 0, run_now = False, **kwargs):

```

Comment:
please use the same signature as the submit method in the concurrent lib. The additional job-specific policy parameters can be defined in a separate parameter, called job_config, which you can define as dict